### PR TITLE
[WIP] DO NOT MERGE! Added IsDebug to CakeRuntime.

### LINKED ISSUE #902: Add IsDebug to ICakeEnvironment
Add runtime info to ICakeEnvironment to know if performing debug.

### PR DESCRIPTION
**Pull request by mholo65**
Resolves #902

### CODE REVIEW

File: src/Cake.Core/CakeRuntime.cs [hunk 1]
```diff
@@ -17,15 +17,62 @@ public sealed class CakeRuntime : ICakeRuntime
...
+        /// Gets a value indicating whether the current execution is running in debug mode.
+        /// </summary>
+        public bool IsDebug { get; private set; }
+
+        /// <summary>
         /// Initializes a new instance of the <see cref="CakeRuntime"/> class.
         /// </summary>
-        public CakeRuntime()
+        /// <param name="arguments">The arguments.</param>
+        public CakeRuntime(ICakeArguments arguments)
```
Comment (mholo65) [inline src/Cake.Core/CakeRuntime.cs]: This does not work as ICakeArguments is resolved much later on... One solution could be to wrap it around Lazy<T> or Func<T> which is both supported by Autofac for these "chicken-egg" issues..

Another way could be to extract some of the Core classes to a separate module which could be built after the Arguments module..

Thought? @gep13 @patriksvensson

**Comment by gep13** [comment]
@mholo65 just wanted to touch base on this one again...

Where are we with it?  Thanks!

**Comment by mholo65** [comment]
@gep13 as mentioned above, this is a "chiclen-egg" issue with some registrations in Autofac. Need to think of an solution which doesn't require major refactoring.

**Comment by mholo65** [comment]
@patriksvensson 

It's not possible to inject **ICakeArguments** to **CakeRuntime**... For that to work, we'll need to register the **ArgumentsModule** prior to **CoreModule**. Don't even know if that's possible, haven't dug that deep..

Should we resort to implementation as  presented [here](https://github.com/mholo65/cake/blob/2c873ac7845531455646801fbce3f9e83efd6d74/src/Cake.Core/CakeEnvironment.cs#L71)

**Comment by mholo65** [comment]
@patriksvensson ok! I'll fix it. In the meantime, you might want to move #902 and #876 to v0.18.0 milestone

**Comment by gep13** [comment]
>@mholo65 said...
you might want to move #902 and #876 to v0.18.0 milestone

I have cleared those issues from having a milestone assigned.  We can assign them once we know when they are going to be ready.

**Comment by mholo65** [comment]
@patriksvensson still want this and #965?

I could rebase and fix, otherwise close.

**Comment by mholo65** [comment]
Closing. No need for this.

**Pull request closed without merging**